src/main/java/org/apache/commons/lang3/time/StopWatch.java

```diff
@@ -671,4 +677,68 @@ public void unsplit() {
         splitState = SplitState.UNSPLIT;
     }
 
+    <T, E extends Throwable> T get(FailableSupplier<T, E> supplier) throws E {
+        if (isStopped()) {
+            start();
+        } else if (isSuspended()) {
+            resume();
+        }
+
+        try {
+            return supplier.get();
+        } finally {
+            suspend();
+        }
+    }
+
+
+    /**
+     * Take the time of the execution of a given {@linkplain Function}.
+     *
+     * <p>
+     * <b>Take the time of given {@linkplain Function}</b>
+     * <pre>{@code
+     * final StopWatch watch = StopWatch.create();
+     *
+     * String result = watch.apply(it -> it.toLowerCase(Locale.ROOT)).apply("A");
+     * }</pre>
+
+     * <b>Take the time of an applied {@linkplain Function} in a stream</b>
+     *
+     * <pre>{@code
+     * final StopWatch watch = StopWatch.create();
+     *
+     * String result = Stream.of("A", "B", "C")
+     *                       .map(watch.apply(it -> it.toLowerCase(Locale.ROOT)))
+     *                       .collect(Collectors.joining());
+     * }</pre>
+     *
+     * @param function the function those application should be measured
+     *
+     * @return the given function prepared to take time if applied
+     *
+     * @throws IllegalStateException if the StopWatch is not stopped or suspended
+     *
+     * @param <T> the type of the input to the function
+     * @param <R> the type of the result of the function
+     *
+     * @since 3.16
+     */
+    public <T, R> Function<T, R> apply(Function<T, R> function) {
```

Comment:
Why is the new method above failable and this one not?

Reply:
Hi @garydgregory I don't get the question.

Reply:
The argument to `get()` above is a `FailableSupplier`. Here, you use a `Function` but not a `FailableFunction`.